versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractTieredCommitsTest.java

```diff
@@ -408,4 +375,65 @@ void recreateDefaultBranch() throws Exception {
       super(databaseAdapter);
     }
   }
+
+  @Test
+  void keyPrefixBasic(
+      @NessieAdapter @NessieAdapterConfigItem(name = "key.prefix", value = "foo")
+          DatabaseAdapter foo,
+      @NessieAdapter @NessieAdapterConfigItem(name = "key.prefix", value = "bar")
+          DatabaseAdapter bar)
+      throws Exception {
+
+    BranchName main = BranchName.of("main");
+
+    foo.commit(
+        ImmutableCommitAttempt.builder()
+            .commitToBranch(main)
+            .commitMetaSerialized(ByteString.copyFromUtf8("meta-foo"))
+            .addPuts(
+                KeyWithBytes.of(
+                    Key.of("foo"), ContentsId.of("foo"), (byte) 0, ByteString.copyFromUtf8("foo")))
+            .build());
+    bar.commit(
+        ImmutableCommitAttempt.builder()
+            .commitToBranch(main)
+            .commitMetaSerialized(ByteString.copyFromUtf8("meta-bar"))
+            .addPuts(
+                KeyWithBytes.of(
+                    Key.of("bar"), ContentsId.of("bar"), (byte) 0, ByteString.copyFromUtf8("bar")))
+            .build());
+
+    Hash fooMain = foo.toHash(main);
+    Hash barMain = bar.toHash(main);
+
+    Hash fooBranch = foo.create(BranchName.of("foo-branch"), fooMain);
```

Comment:
nit: probably worth extracting `BranchName.of("foo-branch")` into a local variable for reuse. Same for the `barBranch`